src/libraries/System.Private.Xml/src/System/Xml/XPath/XPathNavigator.cs

```diff
@@ -1519,23 +1515,22 @@ public virtual void InsertBefore(XPathNavigator newSibling)
             {
                 throw new InvalidOperationException(SR.Xpn_BadPosition);
             }
-            XmlReader reader = newSibling.CreateReader();
+            using XmlReader reader = newSibling.CreateReader();
             InsertBefore(reader);
         }
 
         public virtual void InsertAfter(string newSibling)
         {
-            XmlReader reader = CreateContextReader(newSibling, false);
+            using XmlReader reader = CreateContextReader(newSibling, false);
             InsertAfter(reader);
         }
 
         public virtual void InsertAfter(XmlReader newSibling)
         {
             ArgumentNullException.ThrowIfNull(newSibling);
 
-            XmlWriter writer = InsertAfter();
+            using XmlWriter writer = InsertAfter();
```

Comment:
Are any of these changes actually improving things? It's possible they could lead to performance regressions, as these are mostly methods that could otherwise be inlined (in particular with dynamic PGO), and now with the exception handling they likely won't be. I'd prefer to close this PR unless it's actually solving a real run-time problem.

Reply:
> Are any of these changes actually improving things?

I have no data to make that claim.

> I'd prefer to close this PR unless it's actually solving a real run-time problem.

Fair enough!

Reply:
Ok, thanks